### PR TITLE
Enable Cruise Control to identify the entity that paused/resumed load monitoring

### PR DESCRIPTION
**Pull request by efeg**
Addresses the request in issue https://github.com/linkedin/cruise-control/issues/424.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java [hunk 1]
```diff
@@ -233,7 +233,8 @@ public LoadMonitorState state(OperationProgress operationProgress, MetadataClien
                                         validPartitionRatio,
                                         numValidPartitions,
                                         totalNumPartitions,
-                                        extrapolations);
+                                        extrapolations,
+                                        _loadMonitorTaskRunner.reasonOfLatestPauseOrResume());
       case SAMPLING:
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java]: should we also add the reason information for `SAMPLING` state?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorState.java [hunk 2]
```diff
@@ -200,28 +232,30 @@ public static LoadMonitorState loading(int numValidSnapshotWindows,
...
     float validPartitionPercent = (float) _numValidMonitoredPartitions / _totalNumPartitions;
     float validWindowPercent = (float) _numValidWindows / _monitoredWindows.size();
     switch (_loadMonitorTaskRunnerState) {
       case NOT_STARTED:
         return String.format("{state: %s}", _loadMonitorTaskRunnerState);
 
       case RUNNING:
-      case SAMPLING:
       case PAUSED:
+        return String.format("{state: %s%s, NumValidWindows: (%d/%d) (%.3f%%), NumValidPartitions: %d/%d (%.3f%%), flawedPartitions: %d%s}",
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorState.java]: missing the format field for `_reasonOfLatestPauseOrResume`

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java (hunk 1 above)
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java]: Reason is only relevant for `RUNNING -> PAUSED` and `PAUSED -> RUNNING` transitions. Is there a case that I am missing?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorState.java (hunk 2 above)
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorState.java]: The `%s` at the end is used for `_reasonOfLatestPauseOrResume`.

**Pull request merged**